shell/src/main/java/alluxio/cli/ValidateEnv.java

```diff
@@ -0,0 +1,392 @@
+  private static final ValidationTask WORKER_DATA_VALIDATION_TASK = create(
+      "worker.data.port.available",
+      new PortAvailabilityValidationTask(ServiceType.WORKER_DATA, AlluxioWorker.class));
+  private static final ValidationTask WORKER_RPC_VALIDATION_TASK = create(
+      "worker.rpc.port.available",
+      new PortAvailabilityValidationTask(ServiceType.WORKER_RPC, AlluxioWorker.class));
+  private static final ValidationTask WORKER_WEB_VALIDATION_TASK = create(
+      "worker.web.port.available",
+      new PortAvailabilityValidationTask(ServiceType.WORKER_WEB, AlluxioWorker.class));
+
+  // ssh validations
+  private static final ValidationTask MASTERS_SSH_VALIDATION_TASK = create(
+      "masters.ssh.reachable",
+      new SshValidationTask("masters"));
+  private static final ValidationTask WORKERS_SSH_VALIDATION_TASK = create(
+      "workers.ssh.reachable",
+      new SshValidationTask("workers"));
+
+  private static final Map<String, Collection<ValidationTask>> TARGET_TASKS =
+      initializeTargetTasks();
+
+  private static Map<String, Collection<ValidationTask>> initializeTargetTasks() {
+    Map<String, Collection<ValidationTask>> targetMap = new TreeMap<>();
+    targetMap.put("master", Arrays.asList(
+        MASTER_RPC_VALIDATION_TASK,
+        MASTER_WEB_VALIDATION_TASK,
+        MASTERS_SSH_VALIDATION_TASK,
+        WORKERS_SSH_VALIDATION_TASK
+    ));
+    targetMap.put("worker", Arrays.asList(
+        WORKER_DATA_VALIDATION_TASK,
+        WORKER_RPC_VALIDATION_TASK,
+        WORKER_WEB_VALIDATION_TASK,
+        MASTERS_SSH_VALIDATION_TASK,
+        WORKERS_SSH_VALIDATION_TASK
+    ));
+    targetMap.put("local", TASK_NAMES.keySet());
+    return targetMap;
+  }
+
+  private interface ValidationTask {
+    boolean validate();
+  }
+
+  private static ValidationTask create(String name, ValidationTask task) {
+    TASK_NAMES.put(task, name);
+    return task;
+  }
+
+  private static boolean isLocalPortAvailable(int port) {
+    try (ServerSocket socket = new ServerSocket(port)) {
+      socket.setReuseAddress(true);
+      return true;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  private static boolean isAddressReachable(String hostname, int port) {
+    try (Socket socket = new Socket(hostname, port)) {
+      return true;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  private static boolean isAlluxioRunning(String className) {
+    String[] command = {"bash", "-c",
+                        "ps -Aww -o command | grep -i \"[j]ava\" | grep " + className};
+    try {
+      Process p = Runtime.getRuntime().exec(command);
+      try (InputStreamReader input = new InputStreamReader(p.getInputStream())) {
+        if (input.read() >= 0) {
+          return true;
+        }
+      }
+      return false;
+    } catch (IOException e) {
+      System.err.format("Unable to check Alluxio status: %s.%n", e.getMessage());
+      return false;
+    }
+  }
```

Comment:
Consider doing this in another manner, for example hitting a rest endpoint. This will clean up the dependencies and also not require any hardcoded class names.